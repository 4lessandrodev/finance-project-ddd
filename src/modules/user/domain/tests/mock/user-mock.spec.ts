import UserMock from "./user.mock";

describe('user.mock', () => {
	
	const userMock = new UserMock();

	it('should create a valid user model mock', () => {
		const model = userMock.model();

		expect(model).toBeDefined();
	});

	it('should create a valid user model mock with provided id', () => {
		const model = userMock.model({ id: 'provided_id' });

		expect(model.id).toBe('provided_id');
		expect(model).toMatchSnapshot();
	});

	it('should create a valid user aggregate mock with provided id', () => {
		const domain = userMock.domain({ id: 'provided_id' });


		expect(domain.isSuccess).toBeTruthy();
		expect(domain.getResult().id.uid).toBe('provided_id');
		expect(domain).toMatchSnapshot();
	});

	it('should create a valid user aggregate mock', () => {
		const domain = userMock.domain();

		expect(domain.isSuccess).toBeTruthy();
	});
});
