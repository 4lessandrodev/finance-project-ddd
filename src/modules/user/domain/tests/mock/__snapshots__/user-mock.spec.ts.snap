// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user.mock should create a valid user aggregate mock with provided id 1`] = `
Result {
  "_value": UserAggregate {
    "_domainEvents": Array [],
    "_id": DomainId {
      "isNew": false,
      "props": Object {
        "value": "provided_id",
      },
    },
    "entityName": "UserAggregate",
    "props": Object {
      "ID": DomainId {
        "isNew": false,
        "props": Object {
          "value": "provided_id",
        },
      },
      "createdAt": 2022-01-01T00:00:00.000Z,
      "deletedAt": undefined,
      "email": EmailValueObject {
        "props": Object {
          "value": "valid_email@domain.com",
        },
      },
      "isDeleted": false,
      "password": PasswordValueObject {
        "props": Object {
          "value": "valid_password",
        },
      },
      "terms": Array [
        TermValueObject {
          "props": Object {
            "acceptedAt": DateValueObject {
              "ONE_DAY": 86400000,
              "ONE_HOUR": 3600000,
              "ONE_MINUTE": 60000,
              "ONE_MONTH": 2678400000,
              "ONE_WEEK": 604800000,
              "ONE_YEAR": 31622400000,
              "props": Object {
                "value": 2022-01-01T00:00:00.000Z,
              },
            },
            "ip": IpValueObject {
              "props": Object {
                "value": "127.0.0.1",
              },
            },
            "isAccepted": true,
            "userAgent": Object {
              "name": "firefox",
              "os": "linux",
              "type": "browser",
              "version": "86.0.1",
            },
          },
        },
      ],
      "updatedAt": 2022-01-01T00:00:00.000Z,
    },
  },
  "error": null,
  "isFailure": false,
  "isSuccess": true,
  "statusCode": "OK",
  "statusCodeNumber": 200,
}
`;

exports[`user.mock should create a valid user model mock with provided id 1`] = `
Object {
  "createdAt": 2022-01-01T00:00:00.000Z,
  "email": "valid_email@domain.com",
  "id": "provided_id",
  "isDeleted": false,
  "password": "valid_password",
  "terms": Array [
    Object {
      "acceptedAt": 2022-01-01T00:00:00.000Z,
      "ip": "127.0.0.1",
      "isAccepted": true,
      "userAgent": Object {
        "name": "firefox",
        "os": "linux",
        "type": "browser",
        "version": "86.0.1",
      },
    },
  ],
  "updatedAt": 2022-01-01T00:00:00.000Z,
}
`;
